pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+                    return True
+            else:
+                if varGenPos <= exonStart and varGenPos >= exonEnd:
+                    return True
+        return False
+    else:
+        return "varOutBounds"    
+
+    
+def varInSpliceDonor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice donor region
+    splice donor region = last 3 bases in exon and first 6 bases in intron
+    Returns True if variant in splice donor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    donorBounds = getRefSpliceDonorBoundaries(variant)
+    for exon in donorBounds.keys():
+        donorStart = donorBounds[exon]["donorStart"]
+        donorEnd = donorBounds[exon]["donorEnd"]
+        if varStrand == "+":
+            if varGenPos >= donorStart and varGenPos <= donorEnd:
+                return True
+        else:
+            if varGenPos <= donorStart and varGenPos >= donorEnd:
+                return True
+    return False
+
+
+def varInSpliceAcceptor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice acceptor region
+    splice acceptor region = 20 bases preceding exon and first 3 bases in exon
+    Returns True if variant in splice acceptor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    acceptorBounds = getRefSpliceAcceptorBoundaries(variant)
+    for exon in acceptorBounds.keys():
+        acceptorStart = acceptorBounds[exon]["acceptorStart"]
+        acceptorEnd = acceptorBounds[exon]["acceptorEnd"]
+        if varStrand == "+":
+            if varGenPos >= acceptorStart and varGenPos <= acceptorEnd:
+                return True
+        else:
+            if varGenPos <= acceptorStart and varGenPos >= acceptorEnd:
+                return True
+    return False
+
+
+def varInCiDomain(variant, boundaries):
+    '''
+    Given a variant, determines if variant is in a clinically important domain
+    Second argument determiens which boundaries (ENIGMA or PRIORS) are used for CI domains
+    Returns True if variant in CI domain
+    '''
+    varGenPos = int(variant["Pos"])
+    varGene = variant["Gene_Symbol"]
+    inExon = varInExon(variant)
+    if varGene == "BRCA1":
+        for domain in brca1CIDomains[boundaries].keys():
+            domainStart = brca1CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca1CIDomains[boundaries][domain]["domEnd"]
+            if varGenPos <= domainStart and varGenPos >= domainEnd and inExon == True:
+                return True
+    elif varGene == "BRCA2":
+        for domain in brca2CIDomains[boundaries].keys():
+            domainStart = brca2CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca2CIDomains[boundaries][domain]["domEnd"]
+            if varGenPos >= domainStart and varGenPos <= domainEnd and inExon == True:
+                return True
```

Comment:
check

Reply:
Same as above but for minus strand instead of plus strand transcripts

Reply:
same as above with the checkWithinBoundaries function